model/task/generate_tasks_estimations.go

```diff
@@ -0,0 +1,72 @@
+package task
+
+import (
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/utility"
+	adb "github.com/mongodb/anser/db"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+const (
+	numRevisionsToSearch = 10
+)
+
+func (t *Task) setGenerateTasksEstimations() error {
+	if !t.GenerateTask || (t.EstimatedNumGeneratedTasks != nil && t.EstimatedNumActivatedGeneratedTasks != nil) {
+		return nil
+	}
+
+	query := db.Query(
+		ByPreviousCommit(
+			t.BuildVariant,
+			t.DisplayName,
+			t.Project,
+			evergreen.RepotrackerVersionRequester,
+			t.RevisionOrderNumber,
+		),
+	).Sort([]string{"-" + RevisionOrderNumberKey}).Limit(numRevisionsToSearch)
+
+	tasks := []Task{}
+	err := db.FindAllQ(Collection, query, &tasks)
+	if adb.ResultsNotFound(err) {
```

Comment:
nit: if `adb.ResultsNotFound` and `err != nil` are both return cases, you can just consolidate them into one `err != nil` return since `adb.ResultsNotFound` is just a special case error.

I also thought `adb.ResultsNotFound` was only returned for find one queries, not find all? I could be wrong on that, but worth checking if this is a case that could never happen.